### PR TITLE
shtools: update to 4.8, and switch from Accelerate framework to openblas

### PR DESCRIPTION
**Pull request by MarkWieczorek**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H15
Xcode 12.3 12C33

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: science/shtools/Portfile [hunk 1]
```diff
@@ -36,10 +36,10 @@ variant openmp description {Add OpenMP support} {
...
 test.target         fortran-tests-no-timing
-test.args           PREFIX=${prefix} F95="${prefix}/bin/gfortran-mp-10" LAPACK="-framework Accelerate" BLAS=""
+test.args           PREFIX=${prefix} F95="${prefix}/bin/gfortran-mp-10" LAPACK="-lopenblas" BLAS=""
 
 destroot.args       PREFIX=${prefix}
 
 depends_build       port:gcc10
 depends_test        port:gcc10
-depends_lib         port:fftw-3
+depends_lib         port:fftw-3 port:openblas
```
Comment (mf2k) [inline science/shtools/Portfile:45]: Please put this new dependency on openblas on the next line.

**Review by mf2k — CHANGES_REQUESTED** [review]

File: science/shtools/Portfile (hunk 1 above)
Comment (MarkWieczorek) [reply science/shtools/Portfile]: thanks! done...

**Review by mf2k — APPROVED** [review]
In the commit message, you should mention switching from the Accelerate framework to openblas.

**Comment by kencu** [comment]
there are quite a few thing in this Portfile that are not being done in the usual MacPorts way...

**Comment by MarkWieczorek** [comment]
Hi @kencu, I'd be happy to make changes. Could you let me know what the problems are?

**Comment by kencu** [comment]
First off, this seems to work, so we can commit it.

(Aside - why are we changing from Accelerate to openBLAS?)

this port hard-codes gcc-10 and the compiler into the build, but we don't usually do it that way. For example, some OS versions can't use gcc-10, eg Apple Silicon, so it is broken for all new users. So what we do instead is use a compiler selection mechanism that tells MacPorts what the port needs, and then MacPorts works out the right compiler for the given software on a given OS.

There are built-in mechanisms in base for most things, and then a "compilers" portgroup that adds more compiler selection tools for things that need fortran, etc.

Also, there is a "linear algebra" portgroup that I have used occasionally that has functionality to work out certain features that seem relevant to this port.

**Comment by MarkWieczorek** [comment]
For openblas, this is just to avoid using the default version that comes with macOS. The homebrew project advises to use openblas instead of the default, so I figured that I should do the same with macports. I seem to recall that there is some kind of bug in the default version.

As for the hardcoding gcc and and the gfortran compiler name: I will definitely look into how to do this better. If there is a particular project that you know of, that would help me to at least get started. This will likely take me a while, so either you can merge this PR, or people can continue using the last release (which unfortunately has the same problem...). Thanks for the comments!

**Pull request merged**